### PR TITLE
Add se.manyver.Manyverse

### PR DESCRIPTION
**Pull request by atomrc**
Adds the manyverse app to flathub (see https://www.manyver.se/)

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: se.manyver.Manyverse.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+rename-desktop-file: manyverse.desktop
+rename-icon: manyverse
+finish-args:
+  - --socket=x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --share=network
+  - --allow=bluetooth
+  # full filesystem access is needed in order for the user to be able to browse and upload files from their filesystem
+  - --filesystem=home:ro
```
Comment (atomrc) [inline se.manyver.Manyverse.yaml:18]: We need a ro access to the home directory in order for the user to be able to browse his own files when uploading something to the application

**Comment by atomrc** [comment]
bot, build se.manyver.Manyverse

**Comment by staltz** [comment]
Hi FlatHub maintainers, I just want to note that @atomrc is part of our team, so  his PR qualifies as a "maintainer".

If/when this is accepted, could you also add me @staltz to the newly created github repo?

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{    
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: I had comments: why is this x86_64 only?

File: se.manyver.Manyverse.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: manyverse
+separate-locales: false
+rename-desktop-file: manyverse.desktop
+rename-icon: manyverse
+finish-args:
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline se.manyver.Manyverse.yaml:13]: does it support wayland?

Also even as fallback `--share=ipc` is needed for x11

File: se.manyver.Manyverse.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  - --socket=pulseaudio
+  - --share=network
+  - --allow=bluetooth
+  # full filesystem access is needed in order for the user to be able to browse and upload files from their filesystem
+  - --filesystem=home:ro
+  - --filesystem=xdg-config/manyverse
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.portal.Background
+  - --own-name=org.kde.*
```
Comment (hfiguiere) [inline se.manyver.Manyverse.yaml:22]: this is a very broad dbus own.

File: flathub.json (hunk 2 above)
Comment (atomrc) [reply flathub.json:2]: This constraint might indeed not be needed.

File: se.manyver.Manyverse.yaml (hunk 3 above)
Comment (atomrc) [reply se.manyver.Manyverse.yaml:13]: This is a part where, honestly, I am not actually sure. It's an electron app, does it make sense to add the wayland socket for an electron app, I am not sure. I'd appreciate any suggestion there :)

File: se.manyver.Manyverse.yaml (hunk 4 above)
Comment (atomrc) [reply se.manyver.Manyverse.yaml:22]: That's also good point, I might not even need an own name. How do I know exactly if I need a own name?

**Comment by atomrc** [comment]
@hfiguiere Thanks for you review. Quick question, shall I make the changes in the newly create repo now?

File: se.manyver.Manyverse.yaml (hunk 3 above)
Comment (hfiguiere) [reply se.manyver.Manyverse.yaml:13]: in general Electron doesn't support wayland.

**Comment by hfiguiere** [comment]
yes. nothing else will happen here.

**Comment by atomrc** [comment]
It seems that the build is currently failing on flathub. I am not sure why it seems to be at the upload step with the error 
`Api call to https://hub.flathub.org/api/v1/build/84342/upload failed with status 408, details: {'status': 408, 'error-type': 'no-error', 'message': 'No json error details from server'}`

here https://flathub.org/builds/#/builders/35/builds/10346

Is it something I need to fix on my end?

**Comment by atomrc** [comment]
@hfiguiere I have made the changes to the manyverse repo. But we seem to have deployment errors on the build script. 
Seems like the build for aarch64 architecture is hanging https://flathub.org/builds/#/builders/32/builds/86510

Any idea what is happening?

**Comment by hfiguiere** [comment]
there is no source for aarch64.... but it'll try to build it. (the question was not about removing the flathub.json but why this wasn't available on aarch64, and it seems there never was an answer)

**Comment by hfiguiere** [comment]
technically the build failed because `ar: manyverse*.deb: No such file or directory` on aarch64.

**Comment by atomrc** [comment]
Right, makes sense. Sorry I got a little confused. So yeah we do not have, on manyverse side, a job that builds an aarch64 binary.
Until then it's probably safer to only have the x86 build, WDYT?

**Comment by hfiguiere** [comment]
why not building the flatpak from source?

**Comment by staltz** [comment]
> why not building the flatpak from source?

Initially, another contributor did it from the .deb and we kept that method because it comes with the `.desktop` file ready.

About aarch64, it might be challenging to build because we have node.js native modules (leveldown, sodium-native, etc) which may fail compilation for various reasons. We want to support aarch64 eventually, but maybe it's not worth forcing it at this moment. We can work iteratively.

**Comment by hfiguiere** [comment]
> > why not building the flatpak from source?
> 
> Initially, another contributor did it from the .deb and we kept that method because it comes with the `.desktop` file ready.

If you don't manage the .destkop in you source repository it's already a bug.

> 
> About aarch64, it might be challenging to build because we have node.js native modules (leveldown, sodium-native, etc) which may fail compilation for various reasons. We want to support aarch64 eventually, but maybe it's not worth forcing it at this moment. We can work iteratively.

You can start by building from source here.

**Comment by staltz** [comment]
Yup, we have hundreds of bugs, this is just one more. Have to prioritize them, and this .desktop management one is not the worst bug.

**Pull request closed without merging**